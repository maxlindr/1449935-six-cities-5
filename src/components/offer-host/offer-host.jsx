import React from 'react';
import PropTypes from 'prop-types';
import {hostPropTypes} from '../../prop-types/prop-types';

function OfferHost(props) {
  const {host, description} = props;

  const avatarWrapperClassname = host.super
    ? `property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper`
    : `property__avatar-wrapper user__avatar-wrapper`;

  return (
    <div className="property__host">
      <h2 className="property__host-title">Meet the host</h2>
      <div className="property__host-user user">
        <div className={avatarWrapperClassname}>
          <img
            className="property__avatar user__avatar"
            src={host.avatar}
            alt="Host avatar"
            width={74}
            height={74}
          />
        </div>
        <span className="property__user-name">{host.name}</span>
      </div>
      <div className="property__description">
        {description.split(`\n`).map((paragraph, i) => (
          <p key={i} className="property__text">
            {paragraph}
          </p>
        ))}
      </div>
    </div>
  );
}

OfferHost.propTypes = {
  host: hostPropTypes.isRequired,
  description: PropTypes.string.isRequired
};

export {OfferHost};
export default React.memo(OfferHost);
