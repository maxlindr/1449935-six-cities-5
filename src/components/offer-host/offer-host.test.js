import React from 'react';
import renderer from 'react-test-renderer';
import {OfferHost} from './offer-host';

const HOST = {
  avatar: `img/avatar-angelina.jpg`,
  name: `Host Name`,
  super: true
};

test(`OfferHost должен корректно отображаться`, () => {
  const component = renderer.create(
      <OfferHost
        host={HOST}
        description={`Description`}
      />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
