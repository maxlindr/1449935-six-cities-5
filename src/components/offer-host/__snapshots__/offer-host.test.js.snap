// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferHost должен корректно отображаться 1`] = `
<div
  className="property__host"
>
  <h2
    className="property__host-title"
  >
    Meet the host
  </h2>
  <div
    className="property__host-user user"
  >
    <div
      className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
    >
      <img
        alt="Host avatar"
        className="property__avatar user__avatar"
        height={74}
        src="img/avatar-angelina.jpg"
        width={74}
      />
    </div>
    <span
      className="property__user-name"
    >
      Host Name
    </span>
  </div>
  <div
    className="property__description"
  >
    <p
      className="property__text"
    >
      Description
    </p>
  </div>
</div>
`;
